dataedit/static/metaedit/metaedit.js

```diff
@@ -2,363 +2,412 @@
-
-        // TODO catch init error
-      });
     }
-  })();
 
-  return config;
-};
+    (function init() {
+
+        $('#metaedit-loading').removeClass('d-none');
+
+        config.form = $('#metaedit-form');
+
+        // check if the editor should be initialized with metadata from table or as standalone withou any initial data
+        if (config.standalone == false) {
+            $.when(
+                $.getJSON(config.url_api_meta),
+                $.getJSON('/static/metaedit/schema.json'),
+            ).done(function (data, schema) {
+                config.schema = fixSchema(schema[0]);
+                config.initialData = fixData(data[0]);
+
+
+                 /*  https://github.com/json-editor/json-editor */
+                options = {
+                    startval: config.initialData,
+                    schema: config.schema,
+                    theme: 'bootstrap4',
+                    iconlib: 'fontawesome5',
+                    mode: 'form',
+                    compact: true,
+                    remove_button_labels: true,
+                    disable_collapse: true,
+                    prompt_before_delete: false,
+                    object_layout: "normal",
+                    disable_properties: false,
+                    disable_edit_json: true,
+                    disable_array_delete_last_row: true,
+                    disable_array_delete_all_rows: true,
+                    disable_array_reorder: true,
+                    array_controls_top: true,
+                    no_additional_properties: true,
+                    required_by_default: false,
+                    remove_empty_properties: true, // don't remove, otherwise the metadata will not pass the validation on the server
+                }
+
+                config.editor = new JSONEditor(config.form[0], options);
+
+                /* patch labels */
+                var mainEditBox = config.form.find('.je-object__controls').first();
+                mainEditBox.find('.json-editor-btntype-save').text('Apply');
+                mainEditBox.find('.json-editor-btntype-copy').text('Copy to Clipboard');
+                mainEditBox.find('.json-editor-btntype-cancel').text('Close');
+                mainEditBox.find('.json-editor-btntype-editjson').text('Edit raw JSON');
+
+                bindButtons();
+
+                convertDescriptionIntoPopover();
+                // check for new items in dom
+                (new MutationObserver(function (_mutationsList, _observer) {
+                    convertDescriptionIntoPopover()
+                })).observe(config.form[0], { attributes: false, childList: true, subtree: true });
+
+                // all done
+                $('#metaedit-loading').addClass('d-none');
+                $('#metaedit-icon').removeClass('d-none');
+                $('#metaedit-controls').removeClass('d-none');
+
+                // TODO catch init error
+
+
+                window.JSONEditor.defaults.callbacks = {
```

Comment:
This is your code 👍

Reply:
We have to add this code to the else in block starting in line 356 - the metaEditor can be initialized as so called "standalone" or via the "edit" button for each table - bot should be able to use the OAT. Maybe we can create a function to aviode more duplicated code?